### DISCUSSION TITLE
When you clone a JsValue, is it swallow or deep clone?

### DISCUSSION THREAD

**Discussion by oblique** [wasm-bindgen/wasm-bindgen#3472] [Q&A] [answered]
When you clone a JsValue, is it swallow or deep clone?

To be more specific, when you clone [web_sys::ReadableStreamDefaultReader](https://docs.rs/web-sys/latest/web_sys/struct.ReadableStreamDefaultReader.html), is it swallow or deep clone?

**Comment by Liamolucko** [comment, marked as answer]
It's a shallow clone; it just creates a new reference to the same value.
